tools/clang/lib/AST/HlslTypes.cpp

```diff
@@ -68,6 +68,29 @@ bool IsHLSLVecMatType(clang::QualType type) {
   return false;
 }
 
+clang::RecordDecl *getRecordTypeFromParameterType(clang::QualType ParamTy) {
+
+  if (ParamTy->isStructureOrClassType()) {
+    if (const CXXRecordDecl *CXXRD =
+            ParamTy.getCanonicalType()->getAsCXXRecordDecl()) {
+
+      if (const ClassTemplateSpecializationDecl *templateDecl =
+              dyn_cast<ClassTemplateSpecializationDecl>(CXXRD)) {
+
+        auto &TemplateArgs = templateDecl->getTemplateArgs();
+        DXASSERT(TemplateArgs.size() == 1,
+                 "Input record types need to have one template argument");
```

Comment:
The type passed in might not be an "Input record type", it could be a NodeOutput or even a NodeOutputRecord type.

We should probably also have an assert at the beginning of this function that the type passed in is a node object type, so it isn't misused for another type.